### ISSUE TITLE
Support Mending enchantment

### ISSUE THREAD

**Issue by magneticflux-** [ldtteam/minecolonies#1319]
### Minecolonies version

- Version 0.8.3167, MC 1.12, Forge 14.21.1.2424

### Expected behavior

- Fisherman's rod doesn't break due to the XP collected during fishing

### Actual behaviour

- The rod deteriorates at the normal rate, as if Mending wasn't applied

### Steps to reproduce the problem

1. Get a fishing rod with the mending enchantment
2. Give to fisherman and watch him fish
3. Observe that the durability does not go back up after reeling in a fish and collecting experience

**Comment by marchermans** [comment]
This is being worked on. It is not that simple as its sounds so we will do our best.

**Comment by Asherslab** [comment]
when i looked at it, it seemed pretty easy to do, i've asked a new guy to give it a crack 😄

**Comment by magneticflux-** [comment]
@Asherslab Thanks for the quick response! Now that I think about it, something similar should also probably happen with the miner, farmer, guards, blacksmith(?), and baker(?), since they all have opportunities to earn Vanilla Minecraft experience by smelting, fishing, mining, or killing things. Here's the full list of activities that produce XP orbs: [Link](http://minecraft.gamepedia.com/Experience#Experience_amounts_by_source).

EDIT: The relevant code for repairing is done in the `EntityXPOrb` class, line `#237`, in the `onCollideWithPlayer` method.

**Comment by marchermans** [comment]
Yeah that is what i was worried about
The actual repair code is not in the Enchantment
But in the XPOrb.

**Comment by Electrocutor** [comment]
Perhaps if you can figure out a way to route this it could also allow all of the Tinkers effects to also work on NPCs.

**Comment by Asherslab** [comment]
@Electrocutor any chance you could make a seperate issues and expand on what you mean by that?

**Comment by ghost** [comment]
Well, I fixed this issue. The fix will probably be in the next release or so.

**Comment by magneticflux-** [comment]
@Vitridax I'm running version 1.12-0.8.3302 and the fisherman isn't repairing his rod using experience. In the fisherman AI class, it seems to talk about collection XP orbs in a comment. Maybe the fisherman isn't using the usual method of adding XP? My fisherman is level 17, so he's still gaining XP somehow.

**Comment by ghost** [comment]
@magneticflux- The code is in the Citizen class, not the fisherman class. For some reason (unknown to me atm) fishing rods do not work, but other tools do. I just tested version 1.10.2-0.8.3265 (the code is identical between versions). A fisherman could not mend a fishing rod, but a builder could mend a shovel. I will look into this more later.

**Comment by ghost** [comment]
There is another bug that I found that may also be your problem. If a citizen's level is maxed out for their hut (lv 2,4,8,16 for hut lv 1,2,3,4), they will not mend tools either. I don't think this is your problem, but you might find it later and think it is the same thing, so I thought I'd mention it.

**Comment by magneticflux-** [comment]
@Vitridax My fisherman's hut is level 4, but he's level 17 for some reason... I might be running into both bugs. None of my other workers are that high level right now.

**Comment by Raycoms** [comment]
@Vitridax are you still working on this?

**Comment by magneticflux-** [comment]
What's the status of this? I looked through where XP flows and this is the path I traced out:

1. `EntityAIWorkFisherman` creates and `EntityFishHook` when ready to fish
2. `EntityFishHook` is cast and reeled in and spawns the appropriate `EntityXPOrb`s
3. `EntityCitizen` calls `gatherXpOrbs()` regularly and picks up the XP from fishing
4. `gatherXpOrbs()` deletes the `EntityXpOrb`s and adds half the value using the `EntityCitizen`'s `addExperience(int)` method
5. The `addExperience(int)` method calls `applyMending(int)` and removes the appropriate amount of XP

I'm not seeing any bugs, so it looks like a weird edge case that is only triggered by the Fisherman.

EDIT: I was looking at the real EntityFishHook, not the custom one. The custom one still spawns an `EntityXPOrb` though.
